### DISCUSSION TITLE
War Support

### DISCUSSION THREAD

**Discussion by spacepost** [oracle/coherence-operator#490] [Q&A]
Is there any way to get this to work for a pre-packaged application that uses coherence clustering that is hosted in the websphere-traditional or tomcat docker images?

**Comment by rstafford** [comment]
Could you please indicate what the pre-packaged application is?

**Comment by thegridman** [comment]
No, the Coherence Operator cannot manage other types of server like WebSphere, Tomcat etc. The Coherence Operator is specifically for stand-alone Coherence applications. The operator can manage Coherence cluster members that are use frameworks like Helidon, Spring, Micronaut that use embedded web servers, but not the other way around and particularly not where Coherence is in a .war or .ear. 

In theory running an image that contains some kind of web-server where Coherence is embedded in a .war should be possible. We have never tried or tested this though and it would certainly require some changes to the operator to get it working.

**Comment by rjeberhard** [comment]
Would it be possible for you to use WebLogic rather than WebSphere or Tomcat? If so, we have a separate operator for WebLogic, which does support managed Coherence clusters and Java EE applications.

**Reply by spacepost** [reply]
It's possible that we might be able to use WebLogic. I know that the application supports it. What could be problematic is our licensing situation for that. Websphere containers were the obvious choice at the time as we were and still are in some cases hosting the application using Websphere on VM's, so by using the same product we could fairly easily lift and shift the deployment to the containers with minimal changes and even share things like our secure configuration templates.

The application in question is Oracle's Insurance Policy Administration (OIPA) and we are currently hosting it using the ibmcom/websphere-traditional:latest (9.0.5.8) image.
